### ISSUE TITLE
"java.lang.IllegalArgumentException: Can't find sexp boundary!" (with reduced test-case)

### ISSUE THREAD

**Issue by Peeja** [clojure-emacs/refactor-nrepl#115]
https://github.com/Peeja/refactor-test

To reproduce, open `refactor-test/src/refactor_test/core.clj` and try to `cljr-find-usages`. I get a `java.lang.IllegalArgumentException: Can't find sexp boundary!`. This should happen from any source file in the project.

The issue appears to be a bug in `refactor-nrepl.util/search-sexp-boundary`. You can get the same error by calling

```clojure
(in-ns 'refactor-nrepl.util)
(search-sexp-boundary (partial < 0) :backwards "(defn foo \")\" [])")
```

which is what `refactor-nrepl.util/get-last-sexp` is calling.

**Comment by Peeja** [comment]
I describe in the test case what the problem is, but I should say here too: It doesn't like strings which contain close-parens. We have a docstring with an ordered list in it, which gums up the whole works.

**Comment by benedekfazekas** [comment]
thx for the bugreport @Peeja will look into this

**Comment by Peeja** [comment]
Also, it looks to me like `refactor-nrepl.util/get-last-sexp` returns the *first* N characters of the given string, where N is the length of the *last* sexp in the string. I would expect it to return the *last* N characters, which would be, well, the last sexp in the string. That is,

```clojure
(get-last-sexp "(foo bar) (1 2)") ; => "(foo ", rather than "(1 2)"
```

But maybe there's some hidden brilliance at work here? That doesn't appear to be an edge case, just the current "normal" behavior of the function, and all the existing tests pass, so I could be missing something.

**Comment by benedekfazekas** [comment]
> some hidden brilliance

haha, maybe just a big fat bug tbh. ta for the really thorough testing of this...

**Comment by expez** [comment]
This also goes for unbalanced delimiters in comments

**Comment by expez** [comment]
This fix will be out in the `2.0-SNAPSHOT` only.  The big version jump is because we're not forced into depending on clj 1.7 due to reader macros.

I expect to publish that snapshot tomorrow, along with a bump of clj-refactor.

**Comment by expez** [comment]
2.0.0-SNAPSHOT is now the latest version for clj-refactor and refactor-nrepl
